app/src/features/surveys/components/StudyAreaForm.tsx

```diff
@@ -93,11 +113,26 @@ const StudyAreaForm = () => {
     });
 
     // set values
-    setFieldValue('locations', data);
+    setFieldValue('locations', data, true);
   };
 
   return (
     <form onSubmit={handleSubmit}>
+      <YesNoDialog
+        dialogTitle={`Delete all ${values.locations.length} Study Areas?`}
+        dialogText="Are you sure you want to remove these study areas? This action can be undone by exiting the survey page without saving"
+        yesButtonProps={{ color: 'error' }}
+        yesButtonLabel={'Delete'}
+        noButtonProps={{ color: 'primary', variant: 'outlined' }}
+        noButtonLabel={'Cancel'}
+        open={isDeleteOpen}
+        onYes={() => {
+          setIsDeleteOpen(false);
+          onDeleteAll();
+        }}
+        onClose={() => {}}
```

Comment:
(Jeremy might have an opinion on this, and we might not be consistent on it in the app) When a user clicks outside of the dialog, it calls the onClose. So right now it would do nothing, which is maybe what we want anyways? Or possibly we want it to close?

Reply:
I think historically we always made the `onClose` do the same thing as the `onNo`, which is probably why its not an optional prop. If thats not the case anywhere, we could consider making `onClose` optional, and doing nothing by default.

Reply:
I've updated it to close the dialog when clicking off of the dialog. I think in most places in the app it will close the dialog and take no action.